### PR TITLE
SQLiteStatement added to SQLiteUtils

### PR DESCRIPTION
**Pull request by olmisa**
* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Related PRs or issues

[AB#81470](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/81470)

### CODE REVIEW

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java [hunk 1]
```diff
@@ -5,16 +5,88 @@
...
+        assertTrue(checkTableExists(databaseManager, tableName));
+
+        SQLiteUtils.dropTable(databaseManager.getDatabase(), tableName);
+        assertFalse(checkTableExists(databaseManager, tableName));
+    }
+
+    private boolean checkTableExists(DatabaseManager databaseManager, String tableName) {
+        SQLiteQueryBuilder builder = SQLiteUtils.newSQLiteQueryBuilder();
+        builder.appendWhere("tbl_name = ?");
+        Cursor cursor = databaseManager.getCursor("sqlite_master", builder, new String[]{"tbl_name"}, new String[]{tableName}, null);
```
Comment (annakocheshkova) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java:85]: Format

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java [hunk 2]
```diff
@@ -5,16 +5,88 @@
...
+    @Test
+    public void testTableMethods() {
+        DatabaseManager databaseManager = new DatabaseManager(sContext, DATABASE_NAME, "test.setMaximumSize", 1, mSchema, mock(DatabaseManager.Listener.class));
+
+        String tableName = java.util.UUID.randomUUID().toString();
+        ContentValues contentValues = new ContentValues();
+        contentValues.put("someKey", "someValue");
+        SQLiteUtils.createTable(databaseManager.getDatabase(), tableName, contentValues);
+        assertTrue(checkTableExists(databaseManager, tableName));
+
```
Comment (annakocheshkova) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java:77]:
```suggestion
```

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java [hunk 3]
```diff
@@ -5,16 +5,88 @@
...
     @Test
     public void test() {
         new SQLiteUtils();
         assertNotNull(SQLiteUtils.newSQLiteQueryBuilder());
     }
+
+    @Test
+    public void testTableMethods() {
+        DatabaseManager databaseManager = new DatabaseManager(sContext, DATABASE_NAME, "test.setMaximumSize", 1, mSchema, mock(DatabaseManager.Listener.class));
+
```
Comment (annakocheshkova) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java:71]:
```suggestion
```

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java [hunk 4]
```diff
@@ -5,16 +5,88 @@
...
+        String tableName = java.util.UUID.randomUUID().toString();
+        ContentValues contentValues = new ContentValues();
+        contentValues.put("someKey", "someValue");
+        SQLiteUtils.createTable(databaseManager.getDatabase(), tableName, contentValues);
+        assertTrue(checkTableExists(databaseManager, tableName));
```
Comment (annakocheshkova) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java:76]: Maybe we should check key and value as well? This is the important part of the statement that should be correctly compiled.

Comment (olmisa) [reply sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java:78]: Reworked this part. Here we create the only table with structure. So we have only empty columns. I have added a check for created columns.

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java [hunk 5]
```diff
@@ -5,16 +5,95 @@
...
+        SQLiteUtils.dropTable(databaseManager.getDatabase(), tableName);
+        assertFalse(checkTableExists(databaseManager, tableName));
+    }
+
+    private boolean checkTableExists(DatabaseManager databaseManager, String tableName) {
+        SQLiteQueryBuilder builder = SQLiteUtils.newSQLiteQueryBuilder();
+        builder.appendWhere("tbl_name = ?");
+        Cursor cursor = databaseManager.getCursor("sqlite_master", builder, new String[]{"tbl_name"}, new String[]{tableName}, null);
+        try {
+            return cursor.getCount() > 0;
```
Comment (AnastasiaKubova) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java:94]: We need to check here that cursor is not null

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java [hunk 6]
```diff
@@ -5,16 +5,95 @@
...
+        } finally {
+            cursor.close();
```
Comment (AnastasiaKubova) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java:96]: Add the same check here

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java [hunk 7]
```diff
@@ -5,16 +5,95 @@
...
+    }
+
+    @After
+    public void tearDown() {
+        sContext.deleteDatabase(DATABASE_NAME);
+    }
+
+    private static ContentValues generateContentValues() {
+        ContentValues values = new ContentValues();
+        values.put("COL_STRING_NULL", (String) null);
```
Comment (AnastasiaKubova) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java:59]: Why we need to convert null to string?

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java [hunk 8]
```diff
@@ -5,16 +5,95 @@
...
+    /**
+     * Context instance.
+     */
+    @SuppressLint("StaticFieldLeak")
+    private static Context sContext;
+
+    /**
+     * Database schema.
+     */
+    private static ContentValues mSchema;
```
Comment (AnastasiaKubova) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SQLiteUtilsAndroidTest.java:37]: sSchema

File: CHANGELOG.md [hunk 9]
```diff
@@ -5,6 +5,7 @@
 ### App Center
 
 * **[Fix]** Fix an `IncorrectContextUseViolation` warning when calculating screen size on Android 11.
+* **[Fix]** `SQLiteUtils` now executes sql through `SQLiteStatement`.
```
Comment (MatkovIvan) [inline CHANGELOG.md:8]: Why it's a fix? What expected scenario did not work?

File: CHANGELOG.md [hunk 10]
```diff
@@ -5,6 +5,7 @@
 ### App Center
 
 * **[Fix]** Fix an `IncorrectContextUseViolation` warning when calculating screen size on Android 11.
+* **[Feature]** `SQLiteUtils` now executes sql through `SQLiteStatement`.
```
Comment (MatkovIvan) [inline CHANGELOG.md:8]: Please explain how exactly this affects to the SDK's behaviour. For now it's not clear and I do not see why do we need this change

**Pull request closed without merging**